cookbook/doctrine/registration_form.rst

```diff
@@ -138,230 +170,143 @@ Next, create the form for the ``User`` model::
-        }
-    }
-
-You don't need to use a special method for embedding the ``UserType`` form.
-A form is a field, too - so you can add this like any other field, with the
-expectation that the ``Registration.user`` property will hold an instance
-of the ``User`` class.
-
-Handling the Form Submission
-----------------------------
-
-Next, you need a controller to handle the form. Start by creating a simple
-controller for displaying the registration form::
-
-    // src/Acme/AccountBundle/Controller/AccountController.php
-    namespace Acme\AccountBundle\Controller;
-
-    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
-
-    use Acme\AccountBundle\Form\Type\RegistrationType;
-    use Acme\AccountBundle\Form\Model\Registration;
-
-    class AccountController extends Controller
-    {
-        public function registerAction()
-        {
-            $registration = new Registration();
-            $form = $this->createForm(new RegistrationType(), $registration, array(
-                'action' => $this->generateUrl('account_create'),
-            ));
+                return $this->redirect($redirectUrl);
+            }
 
             return $this->render(
-                'AcmeAccountBundle:Account:register.html.twig',
+                'registration/register.html.twig',
                 array('form' => $form->createView())
             );
         }
     }
 
-And its template:
-
-.. code-block:: html+jinja
-
-    {# src/Acme/AccountBundle/Resources/views/Account/register.html.twig #}
-    {{ form(form) }}
-
-Next, create the controller which handles the form submission. This performs
-the validation and saves the data into the database::
-
-    use Symfony\Component\HttpFoundation\Request;
-    // ...
-
-    public function createAction(Request $request)
-    {
-        $em = $this->getDoctrine()->getManager();
-
-        $form = $this->createForm(new RegistrationType(), new Registration());
-
-        $form->handleRequest($request);
-
-        if ($form->isValid()) {
-            $registration = $form->getData();
-
-            $em->persist($registration->getUser());
-            $em->flush();
-
-            return $this->redirect(...);
-        }
+.. note::
 
-        return $this->render(
-            'AcmeAccountBundle:Account:register.html.twig',
-            array('form' => $form->createView())
-        );
-    }
+    If you decide to NOT use annotation routing (shown above), then you'll
+    need to create a route to this controller:
+    
+    .. configuration-block::
 
-Add new Routes
---------------
+        .. code-block:: yaml
 
-Next, update your routes. If you're placing your routes inside your bundle
-(as shown here), don't forget to make sure that the routing file is being
-:ref:`imported <routing-include-external-resources>`.
+            # app/config/routing.yml
+            user_registration:
+                path:     /register
+                defaults: { _controller: AppBundle:Registration:register }
```

Comment:
Can you please explain why? As I believe this also applies to many other routes (maybe all routes).